test/integration/targets/module_defaults/collections/ansible_collections/testns/testcoll/meta/runtime.yml

```
@@ -1,9 +1,38 @@
 action_groups:
   testgroup:
+    # Test metadata 'extend_group' feature does not get stuck in a recursive loop
+    - metadata:
+        extend_group: othergroup
+    - ping
     - testns.testcoll.echo1
     - testns.testcoll.echo2
```

How about having a regular `metadata` entry in one of the files:
```suggestion
    - testns.testcoll.echo2
    - metadata
```
(Or in some other test)